src/ufuncs.c

```diff
@@ -235,31 +258,52 @@ static void Y_Y_func(char **args, npy_intp *dimensions,
                      npy_intp *steps, void *data)
 {
     FuncGEOS_Y_Y *func = (FuncGEOS_Y_Y *)data;
-    GEOSGeometry *in1, *ret_ptr;
+    GEOSGeometry *in1;
 
-    GEOS_INIT;
+    if ((steps[1] == 0) && (dimensions[0] > 1)) {
```

Comment:
I do not know of any case that this would occur (except for the YY_Y operations). So I am doubting to do this check at all. Still, we might forget something and better to have a Python exception than a segfault or a silent memory leak.

Reply:
Seems OK to leave this in and just note it guards against unexpected behavior.  Always nice to avoid segfaults, though if there is never a case where this would get called, I guess we don't need it...

```
// Fail if inputs do not have the expected structure.
```